src/schemes/fluid/implicit_incompressible_sph/system.jl

```diff
@@ -492,30 +571,11 @@ function pressure_update(system, u, u_ode, semi, time_step)
             density_error[particle] = (new_density - reference_density)
         end
     end
-    avg_density_error = sum(density_error) / nparticles(system)
+    avg_density_error = sum(density_error) / (nparticles(system) * reference_density)
```

Comment:
```suggestion
    relative_density_error = sum(density_error) / (nparticles(system) * reference_density)
```
Now it is a relative error, which is unitless, whereas the density error before had the unit of density.